snooty/parser.py

```diff
@@ -854,6 +862,180 @@ def handle_method_option(self, node: n.Directive) -> None:
+            try:
+                self.check_valid_child(node, child, {"selected-content"})
+                assert isinstance(child, n.ComposableContent)
+                self.handle_composable_content(child, spec_composables)
+                valid_children.append(child)
+
+                # populate parent composable-tutorial with selections used by children
+                for option_key, value_key in child.selections.items():
+                    composable_from_spec = next(
+                        spec_composable
+                        for spec_composable in spec_composables
+                        if spec_composable.id == option_key
+                    )
+                    if not value_key or value_key == "None":
+                        continue
+                    option_from_spec = next(
+                        (
+                            spec_option
+                            for spec_option in composable_from_spec.options
+                            if spec_option.id == value_key
+                        ),
+                        None,
+                    )
+                    composable_option = next(
+                        (
+                            composable_option
+                            for composable_option in composable_options
+                            if composable_option["value"] == option_key
+                        ),
+                        {},
+                    )
+                    if not option_from_spec or not composable_option:
+                        continue
+                    composable_option["selections"] = (
+                        composable_option["selections"] or []
+                    )
+                    if isinstance(composable_option["selections"], list):
+                        composable_option["selections"].append(
+                            {
+                                "value": option_from_spec.id,
+                                "text": option_from_spec.title,
+                            }
+                        )
+
+                default_values_found = default_values_found or all(
+                    child.selections[composable_id] == option_id
+                    for composable_id, option_id in (default_ids_dict.items())
+                )
+
+            except ChildValidationError:
+                continue
+
+        if not default_values_found:
+            self.diagnostics.append(
+                MissingChild(
+                    "composable-tutorial",
+                    "selected-content with selections {default_ids_as_string}",
+                    node.start[0],
+                )
+            )
+        node.composable_options = composable_options
+        node.options = {}
+
+    def handle_composable_content(
+        self,
+        node: n.ComposableContent,
+        spec_composables: List[Composable],
+    ) -> None:
+        selection_ids = re.split(r"\s*,\s*", node.options.get("selections", "").strip())
```

Comment:
(Nit) This operation seems to be used in a few places and it's pretty useful for this string pattern. Is there a way to reuse as a function?